### PR TITLE
Add OpenShift template with OAuth 2.0 authentication

### PR DESCRIPTION
**Pull request by bliemli**

### CODE REVIEW

File: deploy-openshift/kube-ops-view-oauth.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  apiVersion: v1
+  metadata:
+    name: kube-ops-view-oauth
+  stringData:
+    authcode-client-id: "${{KUBE_OPS_VIEW_CLIENT_ID}}"
+    authcode-client-secret: "${{KUBE_OPS_VIEW_CLIENT_SECRET}}"
+parameters:
+  - name: KUBE_OPS_VIEW_APP_DOMAIN
+    description: The hostname to register for the route, leave blank for default
+    value: kube-ops-view.example.com
```
Comment (raffaelespazzoli) [inline deploy-openshift/kube-ops-view-oauth.yaml]: can you make this value mandatory by removing the default (which is not going to be valid in most cases anyway). So the user will have to think about what they want here.

Comment (raffaelespazzoli) [inline deploy-openshift/kube-ops-view-oauth.yaml]: can you make this value mandatory by removing the default (which is not going to be valid in most cases anyway). So the user will have to think about what they want here.

**Review by raffaelespazzoli — CHANGES_REQUESTED** [review]

**Comment by bliemli** [comment]
I removed the default value but provided the example URL in the description so the user knows what the expected format looks like.

**Comment by raffaelespazzoli** [comment]
@bliemli thanks for your contribution.
I am going to accept the PR.
However I think we could improve the user experience of using the template.
1. the KUBE_OPS_APP_VIEW_DOMAIN should be required.
2. the client id and secret I believe can be auto-generated. If not please specify in the doc how they should be created.
would you be willing to create an new PR with these changes?

**Comment by bliemli** [comment]
@raffaelespazzoli Thank you for merging my PR.

> 1. the KUBE_OPS_APP_VIEW_DOMAIN should be required.

This environment variable is not required intentionally. If the template is instantiated without it the route should be created using the default naming scheme of something like <appname>-<projectname>.<default-appdomain> (I don't remember it exactly).

> 2. the client id and secret I believe can be auto-generated. If not please specify in the doc how they should be created.

The client ID and secret have to be created on the OAuth server side, e.g. Keycloak and can't be generated on the client side. I'm going to clarify this in the variables' description.

**Pull request merged**